### PR TITLE
Discard truncated linkified URLs

### LINKED ISSUE #6841: `linkify-code` should not run on truncated links
### Description

`linkify-code` linkifies incomplete (truncated) URLs.

The extension should probably avoid linkifying links that are followed by an ellipsis `…`

### How to replicate the issue + URL

Search for code where GitHub truncates a link

https://github.com/search?q=repo%3AKatsuteDev%2FBackground+marketplace&type=code

[image](https://github.com/search?q=repo%3AKatsuteDev%2FBackground+marketplace&type=code)
↑

### Extension version

23.8.3

### Browser(s) used

Google Chrome 115.0.5790.173

### PR DESCRIPTION
**Pull request by Katsute**
* Closes #6841
 
Simply checks if the linkified URL is followed by `…`

`linkified.lastChild.textContent === '…'`

## Test URLs

* https://github.com/search?q=repo%3AKatsuteDev%2FBackground+marketplace&type=code

#### Before

image

#### After

image

### CODE REVIEW

File: source/github-helpers/dom-formatters.tsx [hunk 1]
```diff
@@ -80,6 +80,10 @@ export function linkifyURLs(element: Element): Element[] | void {
 		},
 	});
 
+	if (linkified.lastChild && linkified.lastChild.textContent === '…') { // Link is followed by …
```
Comment (fregante) [inline source/github-helpers/dom-formatters.tsx:83]:
```suggestion
	if (linkified.lastChild?.textContent === '…') { // Link is followed by …
```

**Comment by fregante** [comment]
I think this breaks any comment that ends with … 

``` js
// Like https://wikipedia.org and others… 
```

**Comment by Katsute** [comment]
It still linkifies if there is a full link and `…`, see the top 3 links in the image

**Comment by Katsute** [comment]
Also works on your comment:

GH (default):

image

RGH:

image

This PR:

image

**Review by fregante — APPROVED** [review]

**Comment by fregante** [comment]
Can you also open an issue on linkify-urls? I'd prefer it there eventually, instead of limiting URLs and then discarding them. 

The current approach also probably  discards multiple URLs if the last one is truncated (rare, though)

**Pull request merged**